src/main/java/net/atos/zac/flowable/TakenService.java

```diff
@@ -168,28 +167,26 @@ public TaskInfo readTask(final String taskId) {
             return task;
         }
         final HistoricTaskInstance historicTaskInstance = findClosedTask(taskId);
-        if (historicTaskInstance != null) {
-            return historicTaskInstance;
+        if (historicTaskInstance == null) {
+            throw new RuntimeException(String.format("No task found with task id '%s'", taskId));
```

Comment:
Most of this code is called by the endpoints to provide data. So it will be handled there and we'll return 500 (instead of 400?)

Added
https://dimpact.atlassian.net/browse/PZ-1778